libcob/intrinsic.c

```diff
@@ -3153,97 +3154,178 @@ cob_switch_value (const int id)
 void
 cob_decimal_pow (cob_decimal *pd1, cob_decimal *pd2)
 {
-	cob_uli_t		n;
-	const int		sign = mpz_sgn (pd1->value);
+	int sign_nbr;
+	int sign_exp;
+	int power_case;
+	int negat_result = 0 ;
 
-	if (unlikely (pd1->scale == COB_DECIMAL_NAN)) {
+	if (unlikely(pd1->scale == COB_DECIMAL_NAN)) {
 		return;
 	}
-	if (unlikely (pd2->scale == COB_DECIMAL_NAN)) {
+	if (unlikely(pd2->scale == COB_DECIMAL_NAN)) {
 		pd1->scale = COB_DECIMAL_NAN;
 		return;
 	}
 
-	if (mpz_sgn (pd2->value) == 0) {
-		/* Exponent is zero */
-		if (sign == 0) {
-			/* 0 ^ 0 */
-			cob_set_exception (COB_EC_SIZE_EXPONENTIATION);
+	sign_nbr = mpz_sgn(pd1->value);
```

Comment:
In order to strictly follow GnuCOBOL's coding style, a space character should be inserted before function calls. And it shouldn't occur before the `;` character.